src/main/java/br/unb/cic/analysis/df/OverridingAssignmentAnalysis.java

```diff
@@ -70,15 +72,20 @@ protected void flowThrough(FlowSet<DataFlowAbstraction> in, Unit u, FlowSet<Data
     @Override
     protected FlowSet<DataFlowAbstraction> gen(Unit u, FlowSet<DataFlowAbstraction> in) {
         FlowSet<DataFlowAbstraction> res = new ArraySparseSet<>();
+
+        filterJInstanceFields(u);
+
         if (isLeftStatement(u) || isRightStatement(u)) {
             Statement stmt = getStatementAssociatedWithUnit(u);
             u.getDefBoxes().forEach(valueBox -> {
                 if(valueBox.getValue() instanceof Local){
                     res.add(new DataFlowAbstraction((Local) valueBox.getValue(), stmt));
                 }else if(valueBox.getValue() instanceof JArrayRef){
-                    res.add(new DataFlowAbstraction((Local) getJArrayRefName((JArrayRef) valueBox.getValue()), stmt));
+                    res.add(new DataFlowAbstraction(containArrayChain(valueBox), stmt));
                 }else if(valueBox.getValue() instanceof  StaticFieldRef) {
                     res.add(new DataFlowAbstraction((StaticFieldRef) valueBox.getValue(), stmt));
+                }else if(valueBox.getValue() instanceof JInstanceFieldRef){
+                    res.add(new DataFlowAbstraction((JInstanceFieldRef) valueBox.getValue(), stmt, generatedChainJInstanceField(valueBox.getValue().toString())));
```

Comment:
You are creating a DataFlowAbstraction with an attribute called chain when it is JInstanceFIeld. But when this JInstanceFIeld doesn't have a chair, what will be put there? the ideal would not be to send the complete chair as the first attribute simply?

Reply:
Good, done! Now it's the same.